scala-xml/src/main/scala/scalaxml/ElemInstances.scala

```diff
@@ -22,12 +26,13 @@ trait ElemInstances {
    * @param parser the SAX parser to use to parse the XML
    * @return an XML element
    */
-  implicit def xml(implicit parser: SAXParser = XML.parser): EntityDecoder[Elem] = {
+  implicit val xml: EntityDecoder[Elem] = {
     import EntityDecoder._
     decodeBy(MediaType.`text/xml`, MediaType.`text/html`, MediaType.`application/xml`){ msg =>
       collectBinary(msg).flatMap[Elem] { arr =>
         val source = new InputSource(new StringReader(new String(arr.toArray, msg.charset.getOrElse(Charset.`US-ASCII`).nioCharset)))
-        try DecodeResult.success(Task.now(XML.loadXML(source, parser)))
+        val saxParser = spf.newSAXParser()
```

Comment:
I think not. This creates a new parser in each request instead of relying on the external parser which is not thread safe